### PR TITLE
Ruby API Flows

### PR DESCRIPTION
**Pull request by khemrajrathore**
- Removed NaiveCalLinker for Ruby
- Upgraded to language engine version 2.0.78
- Increased Dataflow parameter expansion constant from 20 to 100
- Refactor in API Tagger of Ruby

### CODE REVIEW

File: src/main/scala/ai/privado/model/Constants.scala [hunk 1]
```diff
@@ -162,5 +162,5 @@ object Constants {
   val semanticDelimeter = "_A_"
 
   // Ruby defaults
-  val defaultExpansionLimit = 20
+  val defaultExpansionLimit = 100
```
Comment (dbMundada) [inline src/main/scala/ai/privado/model/Constants.scala:165]: What is impact of this change?

File: src/main/scala/ai/privado/languageEngine/ruby/tagger/source/IdentifierTagger.scala [hunk 2]
```diff
@@ -27,12 +27,19 @@ import ai.privado.cache.RuleCache
...
+import io.shiftleft.codepropertygraph.generated.{Cpg, Operators}
 import io.shiftleft.passes.ForkJoinParallelCpgPass
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.*
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
 
 class IdentifierTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
 
+  val cachedCall: List[Call] = cpg.call
```
Comment (dbMundada) [inline src/main/scala/ai/privado/languageEngine/ruby/tagger/source/IdentifierTagger.scala:39]: Where is this used?

File: src/main/scala/ai/privado/model/Constants.scala (hunk 1 above)
Comment (khemrajrathore) [reply src/main/scala/ai/privado/model/Constants.scala:165]: Increase in this limit allows us to generate more tasks in dataflow operation, In dataflow engine the default limit is 1000

File: src/main/scala/ai/privado/languageEngine/ruby/tagger/source/IdentifierTagger.scala (hunk 2 above)
Comment (khemrajrathore) [reply src/main/scala/ai/privado/languageEngine/ruby/tagger/source/IdentifierTagger.scala:39]: Oh, this is not used anywhere. Will need to remove this. 
Will remove this in subsequent PR's as a new commit will again take time for comparison report

**Review by pandurangpatil — APPROVED** [review]
look good to me.

**Pull request merged**